iree/hal/api_string_util_test.cc

```diff
@@ -0,0 +1,1062 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "absl/container/inlined_vector.h"
+#include "absl/strings/string_view.h"
+#include "iree/base/api_util.h"
+#include "iree/base/memory.h"
+#include "iree/base/status.h"
+#include "iree/base/status_matchers.h"
+#include "iree/hal/api.h"
+#include "iree/testing/gtest.h"
+
+namespace iree {
+namespace hal {
+namespace {
+
+using ::iree::testing::status::IsOkAndHolds;
+using ::iree::testing::status::StatusIs;
+using ::testing::ElementsAre;
+using ::testing::Eq;
+
+// TODO(benvanik): move these utils to C++ bindings.
```

Comment:
:P sure are a lot of utils in this `test.cc` file
thankfully they are pretty thin wrappers, but we want to ensure that the test is exercising the underlying C API / impl faithfully

Reply:
yep, this is a testing ground for them - see #1846